internal/service/ec2/vpc_security_group_test.go

```diff
@@ -2057,8 +1965,38 @@ func TestAccVPCSecurityGroup_failWithDiffMismatch(t *testing.T) {
 	})
 }
 
-func TestAccVPCSecurityGroup_ruleLimitExceededAppend(t *testing.T) {
-	ruleLimit := testAccSecurityGroupRulesPerGroupLimitFromEnv()
+var ruleLimit int
+var quotaMutex = &sync.Mutex{}
+
+// testAccSecurityGroup_ruleLimit sets the global "ruleLimit" and is only called once
+// but does not run in parallel slowing down tests. The mutex limits it to one slow down
+// instead of one per test. It cannot run in parallel since it is called by another test
+// and double paralleling is a panic.
+func testAccSecurityGroup_ruleLimit(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ErrorCheck:               acctest.ErrorCheck(t, ec2.EndpointsID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckSecurityGroupDestroy,
+		Steps: []resource.TestStep{
+			// get limit
+			{
+				Config: testAccVPCSecurityGroupConfig_getLimit(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSecurityGroupRuleLimit("data.aws_servicequotas_service_quota.test", &ruleLimit),
+				),
+				PreventPostDestroyRefresh: true, // saves a few seconds
+			},
+		},
+	})
+}
+
+func TestAccVPCSecurityGroup_RuleLimit_exceededAppend(t *testing.T) {
+	quotaMutex.Lock()
+	if ruleLimit == 0 {
+		testAccSecurityGroup_ruleLimit(t)
+	}
+	quotaMutex.Unlock()
```

Comment:
I think these should be `defer quotaMutex.Unlock()` as in non-acceptance test mode `resource.(Parallel)Test` calls `t.Skip()` which ends up exiting the goroutine via [`runtime.Goexit`](https://pkg.go.dev/runtime#Goexit).